component/org.wso2.carbon.identity.dpop/src/main/java/org/wso2/carbon/identity/dpop/validators/DPoPHeaderValidator.java

```diff
@@ -0,0 +1,262 @@
+    }
+
+    private static boolean validateDPoPPayload(String httpMethod, String httpURL, JWTClaimsSet jwtClaimsSet)
+            throws IdentityOAuth2Exception {
+
+        return checkJwtClaimSet(jwtClaimsSet) && checkDPoPHeaderValidity(jwtClaimsSet) && checkJti(jwtClaimsSet) &&
+                checkHTTPMethod(httpMethod, jwtClaimsSet) && checkHTTPURI(httpURL, jwtClaimsSet);
+    }
+
+    private static boolean checkJwk(JWSHeader header) throws IdentityOAuth2ClientException {
+
+        if (header.getJWK() == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("'jwk' is not presented in the DPoP Proof header");
+            }
+            throw new IdentityOAuth2ClientException(DPoPConstants.INVALID_DPOP_PROOF, DPoPConstants.INVALID_DPOP_ERROR);
+        }
+        return true;
+    }
+
+    private static boolean checkAlg(JWSHeader header) throws IdentityOAuth2ClientException {
+
+        JWSAlgorithm algorithm = header.getAlgorithm();
+        if (algorithm == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("'algorithm' is not presented in the DPoP Proof header");
+            }
+            throw new IdentityOAuth2ClientException(DPoPConstants.INVALID_DPOP_PROOF, DPoPConstants.INVALID_DPOP_ERROR);
+        }
+        return true;
+    }
+
+    private static boolean checkHeaderType(JWSHeader header) throws IdentityOAuth2ClientException {
+
+        if (!DPoPConstants.DPOP_JWT_TYPE.equalsIgnoreCase(header.getType().toString())) {
+            if (log.isDebugEnabled()) {
+                log.debug(" typ field value in the DPoP Proof header  is not equal to 'dpop+jwt'");
+            }
+            throw new IdentityOAuth2ClientException(DPoPConstants.INVALID_DPOP_PROOF, DPoPConstants.INVALID_DPOP_ERROR);
+        }
+
+        return true;
+    }
+
+    private static boolean checkJwtClaimSet(JWTClaimsSet jwtClaimsSet) throws IdentityOAuth2ClientException {
+
+        if (jwtClaimsSet == null) {
+            log.debug("'jwtClaimsSet' is missing in the body of a DPoP proof.");
+            throw new IdentityOAuth2ClientException(DPoPConstants.INVALID_DPOP_PROOF, DPoPConstants.INVALID_DPOP_ERROR);
+        }
+        return true;
+    }
+
+    private static boolean checkDPoPHeaderValidity(JWTClaimsSet jwtClaimsSet) throws IdentityOAuth2ClientException {
+
+        Timestamp currentTimestamp = new Timestamp(new Date().getTime());
+        Date issuedAt = (Date) jwtClaimsSet.getClaim(DPoPConstants.DPOP_ISSUED_AT);
+        if (issuedAt == null) {
+            log.debug("DPoP Proof missing the 'iat' field.");
+            throw new IdentityOAuth2ClientException(DPoPConstants.INVALID_DPOP_PROOF, DPoPConstants.INVALID_DPOP_ERROR);
+        }
+
+        boolean isExpired = (currentTimestamp.getTime() - issuedAt.getTime()) > getDPoPValidityPeriod();
+        if (isExpired) {
+            String error = "Expired DPoP Proof";
+            log.debug(error);
```

Comment:
add logs is if(log.isdebugEnabled)

Reply:
Fixed with [a7d4758](https://github.com/kalanika/identity-oauth-addons/commit/a7d47581f7736782add115a9ebd85d0ebee46814).